admin/class-yoast-network-admin.php

```diff
@@ -0,0 +1,248 @@
+	 *
+	 * @return void
+	 */
+	public function handle_update_options_request() {
+		$option_group = filter_input( INPUT_POST, 'network_option_group', FILTER_SANITIZE_STRING );
+
+		check_admin_referer( "$option_group-network-options" );
+
+		$whitelist_options = Yoast_Network_Settings_API::get()->get_whitelist_options( $option_group );
+
+		if ( empty( $whitelist_options ) ) {
+			wp_die( __( 'Sorry, you are not allowed to modify unregistered network settings.', 'wordpress-seo' ), '', 403 );
+		}
+
+		foreach ( $whitelist_options as $option_name ) {
+			$value = null;
+			if ( isset( $_POST[ $option_name ] ) ) {
+				$value = wp_unslash( $_POST[ $option_name ] );
+			}
+
+			WPSEO_Options::update_site_option( $option_name, $value );
+		}
+
+		$settings_errors = get_settings_errors();
+		if ( empty( $settings_errors ) ) {
+			add_settings_error( $option_group, 'settings_updated', __( 'Settings Updated.', 'wordpress-seo' ), 'updated' );
+		}
+
+		$this->persist_settings_errors();
+
+		$sendback = add_query_arg( 'settings-updated', 'true', wp_get_referer() );
+		wp_safe_redirect( $sendback );
+		exit;
+	}
+
+	/**
+	 * Handles a request in which a site's default settings should be restored.
+	 *
+	 * @return void
+	 */
+	public function handle_restore_site_request() {
+		check_admin_referer( 'wpseo-network-restore' );
+
+		$option_group = 'wpseo_ms';
+
+		$site_id = ! empty( $_POST[ $option_group ]['site_id'] ) ? (int) $_POST[ $option_group ]['site_id'] : 0;
+		if ( ! $site_id ) {
+			add_settings_error( $option_group, 'settings_updated', __( 'No site has been selected to restore.', 'wordpress-seo' ), 'error' );
+		}
+		else {
+			$site = get_site( $site_id );
+			if ( ! $site ) {
+				/* translators: %s expands to the ID of a site within a multisite network. */
+				add_settings_error( $option_group, 'settings_updated', sprintf( __( 'Site %d not found.', 'wordpress-seo' ), $site_id ), 'error' );
+			}
+			else {
+				WPSEO_Options::reset_ms_blog( $site_id );
+
+				/* translators: %s expands to the name of a site within a multisite network. */
+				add_settings_error( $option_group, 'settings_updated', sprintf( __( '%s restored to default SEO settings.', 'wordpress-seo' ), esc_html( $site->blogname ) ), 'updated' );
+			}
+		}
+
+		$this->persist_settings_errors();
+
+		$sendback = add_query_arg( 'settings-updated', 'true', wp_get_referer() );
+		wp_safe_redirect( $sendback );
+		exit;
+	}
+
+	/**
+	 * Outputs nonce, action and option group fields for a network settings page in the plugin.
+	 *
+	 * @param string $option_group Option group name for the current page.
+	 *
+	 * @return void
+	 */
+	public function settings_fields( $option_group ) {
+		?>
+		<input type="hidden" name="network_option_group" value="<?php echo esc_attr( $option_group ); ?>" />
+		<input type="hidden" name="action" value="<?php echo esc_attr( self::UPDATE_OPTIONS_ACTION ); ?>" />
+		<?php
+		wp_nonce_field( "$option_group-network-options" );
```

Comment:
I think the `"` aren't needed here

Reply:
Per the WP coding guidelines it was decided at some point to prefer variable interpolation over concatenation, so if the plugin's guidelines in that regard don't differ, I'd rather keep it as is.